src/main/java/com/harrison/messaging/Consumer.java

```diff
@@ -0,0 +1,25 @@
+package com.harrison.messaging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("local")
+public class Consumer {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(Consumer.class);
+    
+    private int count = 1;
+    
+    @RabbitListener(queues = "hello")
+    public void receive(String in) {
+        if (in.equals("49")) {
+            throw new RuntimeException("bad message");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.38.14.86:9000/sonar/coding_rules#rule_key=squid%3AS00112)